pkg/tools/cli/loadtest/usecase/load.go

```diff
@@ -96,7 +103,7 @@ func (r *run) Start(ctx context.Context) (<-chan error, error) {
 			select {
 			case <-ctx.Done():
 				return finalize()
-			case err = <-rech:
+			//case err = <-rech: // TODO: related to #557
```

Comment:
I think the `finalize` is not necessary 🤔 
for example, I substitute the error for the named error in the defer.
```
r.eg.Go(safety.RecoverFunc(func() (err error) {
		defer close(tech)
		defer func() {
			var errs error
			if r.client != nil {
				err = r.client.Close()
				if err != nil {
					errs = errors.Wrap(errs, err.Error())
				}
			}
			err = ctx.Err()
			if err != nil && err != context.Canceled {
				errs = errors.Wrap(errs, err.Error())
			}
			return errs
			}
		}()
		for {
			select {
			case <-ctx.Done():
				return
			//case err = <-rech: // TODO: related to #557
			case err = <-lech:
			}
			if err != nil {
				log.Error(err)
				select {
				case <-ctx.Done():
					return
				case ech <- err:
				}
			}
		}
	}))
```

Reply:
This is based on https://github.com/vdaas/vald/blob/76908b610be98961163ee1b826e797370c5e999e/internal/client/discoverer/discover.go#L137-L155 and some similar lines, so I think it's better to match it.
How about it?